### PR TITLE
Follow-up to Azure auth validation change

### PR DESCRIPTION
**Pull request by DaveCTurner**
Arising from a post-merge review of #111242

### CODE REVIEW

File: modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ long getUploadBlockSize() {
     @SuppressForbidden(reason = "this test uses a HttpHandler to emulate an Azure endpoint")
     private static class AzureBlobStoreHttpHandler extends AzureHttpHandler implements BlobStoreHttpHandler {
         AzureBlobStoreHttpHandler(final String account, final String container) {
-            super(account, container, null /* no auth header validation */);
+            super(account, container, null /* no auth header validation - sometimes it's omitted in these tests (TODO why?) */);
```
Comment (DaveCTurner) [inline modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java:157]: As the (new) comment says, the `Authorization` header is sometimes missing in the calls the SDK makes, but apparently only in this specific test suite. I don't know why. I spent a little time investigating without success, but this is not on the critical path for what I'm actually working towards so I don't want to spend too long on it right now. Definitely worth digging deeper later on.

File: test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java [hunk 2]
```diff
@@ -82,27 +82,27 @@ private boolean isValidAuthHeader(HttpExchange exchange) {
...
     public void handle(final HttpExchange exchange) throws IOException {
         if (isValidAuthHeader(exchange) == false) {
-            try (exchange; var xcb = XContentBuilder.builder(XContentType.JSON.xContent())) {
-                xcb.startObject();
-                xcb.field("method", exchange.getRequestMethod());
-                xcb.field("uri", exchange.getRequestURI().toString());
-                xcb.field("predicate", authHeaderPredicate.toString());
-                xcb.field("authorization", Objects.toString(getAuthHeader(exchange)));
-                xcb.startObject("headers");
+            try (exchange; var builder = XContentBuilder.builder(XContentType.JSON.xContent())) {
```
Comment (DaveCTurner) [inline test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java:85]: Renamed :+1:

File: test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpFixture.java [hunk 3]
```diff
@@ -49,16 +49,22 @@ public enum Protocol {
         HTTPS
     }
 
-    public static Predicate<String> startsWithPredicate(String expectedPrefix) {
+    /**
+     * @param  account The name of the Azure Blob Storage account against which the request should be authorized..
+     * @return a predicate that matches the {@code Authorization} HTTP header that the Azure SDK sends when using shared key auth (i.e.
+     *         using a key or SAS token).
+     * @see <a href="https://learn.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key">Azure docs on shared key auth</a>
```
Comment (DaveCTurner) [inline test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpFixture.java:56]: Hope this helps clarify - the header is coming from the Azure SDK, and I'm working towards a change that should cause it to emit a different kind of auth header, so I want to add these assertions now in order to better support the upcoming change.

Comment (DiannaHohensee) [reply test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpFixture.java:56]: Very helpful 👍

**Review by DiannaHohensee — APPROVED** [review]
Awesome, thanks for circling back! LGTM 👍

**Review by idegtiarenko — APPROVED** [review]

**Pull request merged**